build_projects/shared-build-targets-utils/Publishing/AzurePublisher.cs

```diff
@@ -189,19 +189,19 @@ private static void AutoRenewLease(AzurePublisher instance, string blob, string
 
                 // If the task has been requested to be cancelled, then do so.
                 token.ThrowIfCancellationRequested();
-                
+
                 Thread.Sleep(waitFor);
             }
         }
 
         public void ReleaseLeaseOnBlob(string blob, string leaseId)
         {
+            // Cancel the lease renewal task since we are about to release the lease.
+            ResetLeaseRenewalTaskState();
```

Comment:
Copied from the prior commit:

There'll still be the small window of releasing and renewing at the same time if the renewal is already happening at the time of cancellation, although I'm not certain the best way to handle that. Does .Cancel() block until the thread actually cancels? Perhaps we should also wait for the thread to join when cancelling if not.